src/classes/grade.js

```diff
@@ -1,250 +1,212 @@
-                        message = tosdrMessages.mixed
-                    } else if (tosdrData.score > 0) {
-                        message = tosdrMessages.bad
-                    }
-                }
-
-                result = {
-                    score: tosdrData.score,
-                    class: tosdrData.class,
-                    reasons: {
-                        good: matchGood,
-                        bad: matchBad
-                    },
-                    message: message
-                }
-
-                return result
-            }
-        })
-        return result
-    }
+        // set any values that were passed in
+        attrs = attrs || {}
 
-    /* is the parent site itself a major tracking network?
-     * minus one grade for each 10% of the top pages this
-     * network is found on.
-     */
-    isaMajorTrackingNetwork () {
-        let result = 0
-        if (this.specialPage || !this.domain) return result
-        const parentCompany = utils.findParent(this.domain.split('.'))
-        if (!parentCompany) return result
-        const isMajorNetwork = majorTrackingNetworks[parentCompany.toLowerCase()]
-        if (isMajorNetwork) {
-            result = Math.ceil(isMajorNetwork / 10)
+        if (attrs.hasHttps) {
+            this.setHttps(attrs.hasHttps, attrs.isAutoUpgradeable)
+        }
+        if (attrs.privacyScore) {
+            this.setPrivacyScore(attrs.privacyScore)
+        }
+        if (attrs.parentEntity) {
+            this.setParentEntity(attrs.parentEntity, attrs.prevalence)
+        }
+        if (attrs.trackersBlocked) {
+            Object.keys(attrs.trackersBlocked).forEach((entityName) => {
+                this.addEntityBlocked(entityName, attrs.trackersBlocked[entityName].prevalence)
+            })
+        }
+        if (attrs.trackersNotBlocked) {
+            Object.keys(attrs.trackersNotBlocked).forEach((entityName) => {
+                this.addEntityNotBlocked(entityName, attrs.trackersNotBlocked[entityName].prevalence)
+            })
         }
-        return result
     }
 
-    /*
-     * Calculates and returns a site score
-     */
-    get () {
-        if (this.specialPage) return {}
+    setHttps (hasHttps, isAutoUpgradeable) {
+        // isAutoUpgradeable implies the site has https
+        // make sure hasHttps is correct in case of
+        // e.g. a difference between the two lists
+        this.hasHttps = hasHttps || isAutoUpgradeable
+        this.isAutoUpgradeable = isAutoUpgradeable
+    }
 
-        this.decisions = []
+    setPrivacyScore (score) {
+        this.privacyScore = typeof score === 'number' ? score : UNKNOWN_PRIVACY_SCORE
+    }
 
-        let beforeIndex = 1
-        let afterIndex = 1
+    addEntityBlocked (name, prevalence) {
+        if (!name) return
+        if (!prevalence) prevalence = 1
 
-        this.addDecision({change: 1, index: beforeIndex, why: 'Default grade'})
+        this.entitiesBlocked[name] = prevalence
+    }
```

Comment:
@russellholt following from our discussion about APIs yesterday, I thought a bit more in terms of "what makes sense for the privacy grade algorithm?" rather than copying the existing Score class API.

I realized that since the algorithm doesn't care about individual trackers, it doesn't need to accept individual trackers.

So I've changed the API to e.g.:

```js
grade.addEntityNotBlocked('Oath', 7.06)
grade.addEntityBlocked('comScore', 12.75)
```

I suspect this kind of API might make more sense for the privacy grade calculator, too.